src/main/java/forestry/core/utils/BlockUtil.java

```diff
@@ -228,4 +234,70 @@ private static boolean isVecInsideXZBounds(Vec3 vec, float minX, float minZ, flo
 	private static boolean isVecInsideXYBounds(Vec3 vec, float minX, float minY, float maxX, float maxY) {
 		return vec != null && (vec.xCoord >= minX && vec.xCoord <= maxX && vec.yCoord >= minY && vec.yCoord <= maxY);
 	}
+
+	private enum BlockPickerStyle {
+		VANILLA,
+		BARTWORKS
+	}
+	private final static Map<Class,BlockPickerStyle> pickerForClass = new HashMap<Class,BlockPickerStyle>();
+
+	private static Class cl_BW_MetaGenerated_Blocks;
+	private static Method mthd_BW_GetProperBlock;
+	private static Field fld_BW_mMetaData;
+
+	private static boolean searchedForeignClasses = false;
+	private static void grabForeignClasses() {
+		if (searchedForeignClasses) {
+			return;
+		}
+		try {
+			cl_BW_MetaGenerated_Blocks = Class.forName("com.github.bartimaeusnek.bartworks.system.material.BW_MetaGenerated_Blocks");
+			Class cl_BW_MetaGenerated_Block_TE = Class.forName("com.github.bartimaeusnek.bartworks.system.material.BW_MetaGenerated_Block_TE");
+			for (Method mt : cl_BW_MetaGenerated_Block_TE.getDeclaredMethods()) {
```

Comment:
No point iterating all methods if you know what method you want.. just grab it directly 🤷🏻‍♀️

Reply:
getMethod only works on public ones.

Reply:
So use getDeclaredMethod... don’t need to iterate them all 😂😂